src/middlewared/middlewared/plugins/system.py

```diff
@@ -1359,11 +1381,59 @@ def set_crash_reporting(self):
         CrashReporting.enabled_in_settings = self.middleware.call_sync('system.general.config')['crash_reporting']
 
 
+async def _update_birthday_data(middleware, birthday=None):
+    middleware.logger.debug('Synchornization _update_birthday_data interface')
+    # Check if it is exists already
+    system_obj = (await middleware.call('system.info'))
+    birthday_obj = system_obj['birthday']
+    if birthday_obj is not None:
+        # Already setted before.
+        return
+
+    # If it is not defined yet, it will try to define
+    if birthday is None:
+        birthday = (await middleware.call('system.sync_clock'))
+
+    if birthday is not None:
+        # Update System Settings
+        settings = await middleware.call('datastore.config', 'system.settings')
+        await middleware.call('datastore.update', 'system.settings', settings['id'], {
+            'stg_birthday': birthday,
+        })
+
+
+# Update Birthday Date
+async def _update_birthday(middleware):
+    # Sync clock
+    middleware.logger.debug('Synchronization the clock for system birthday')
+    birthday = None
+    timeout = 3600 * 24
+
+    middleware.register_hook('interface.post_sync', _update_birthday_data)
+
+    while birthday is None:
```

Comment:
I just thought of something we have to handle.

We dont want to set birthday for upgraded systems, only new installs.

Thoughts? @themylogin

Reply:
We can also set it to minimal mtime of files in `/data`; however, chances are that this time would be incorrect.

Reply:
for upgrade systems, are they not driven by the same database without erase?
what would be required to do in those files in `/data`?

Reply:
@bastiao they are, but some files are never touched since the installation, that's why I think minimal mtime will be precise enough. Not sure if it is correct to implement this, as original ticket does not explain what for do we need the birthday date. Thoughts? @william-gr

Reply:
Lets forget about upgrades and only do this for new installs.

For upgrades we will set a value in the database to let us know to not try to set a birthday time.
We will try to set birthday only to new installs.